backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/notification/impl/UserNotificationsConfigRepository.java

```diff
@@ -54,70 +43,37 @@
 
 public class UserNotificationsConfigRepository implements IUserNotificationsConfigRepository
 {
-	private final CCache<UserId, UserNotificationsConfig> userNotificationsConfigsByUserId = CCache.<UserId, UserNotificationsConfig> builder()
-			.tableName(I_AD_User_NotificationGroup.Table_Name)
-			.initialCapacity(100)
-			.cacheMapType(CacheMapType.LRU)
-			.additionalTableNameToResetFor(org.compiere.model.I_AD_User.Table_Name)
-			.additionalTableNameToResetFor(I_AD_NotificationGroup.Table_Name)
-			.build();
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
 
 	@Override
-	public UserNotificationsConfig getByUserId(final UserId adUserId)
+	public void createOrUpdate(@NonNull final UserNotificationGroupCreateRequest request)
 	{
-		return userNotificationsConfigsByUserId.getOrLoad(adUserId, this::retrieveUserNotificationsConfig);
+		final I_AD_User_NotificationGroup userNotificationGroup = CoalesceUtil.coalesceNotNull(loadRecordBy(request.getUserId(),request.getNotificationGroupId()), InterfaceWrapperHelper.newInstance(I_AD_User_NotificationGroup.class));
```

Comment:
pls use coalesceSuppliersNotNull instead of coalesce...

Reply:
done. Reason I initially went with the non-suppliers was that the IWH#newInstance() was not that expensive and I could not foresee any problems with extra calls.

Reply:
that's what the "Consumer economy" adepts want to hear from you, "not needed but i bought it because it was cheap"  :)